{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# find all 15 nDNA proteins (interact with AB) variants in dataset for dbsnp ids \n",
    "# 15 proteins list do not 13 mDNA proteins\n",
    "# results match:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.1.4'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "pd.__version__\n",
    "#'1.1.4'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read trascript uniprot mapping file\n",
    "mtp= pd.read_csv(\"../AB-15_mt-protein-UniProt_RefSeq-nt.txt\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(62, 2) \n",
      "      From              To\n",
      "0  O60506  NM_001159673.1\n",
      "1  O60506  NM_001159674.1\n",
      "2  O60506  NM_001159675.1\n",
      "3  O60506  NM_001159676.1\n",
      "4  O60506  NM_001159677.1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O60506</td>\n",
       "      <td>NM_001159673.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O60506</td>\n",
       "      <td>NM_001159674.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O60506</td>\n",
       "      <td>NM_001159675.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O60506</td>\n",
       "      <td>NM_001159676.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O60506</td>\n",
       "      <td>NM_001159677.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB    refseq_nt_id\n",
       "0    O60506  NM_001159673.1\n",
       "1    O60506  NM_001159674.1\n",
       "2    O60506  NM_001159675.1\n",
       "3    O60506  NM_001159676.1\n",
       "4    O60506  NM_001159677.1"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(mtp.shape, '\\n', mtp.head())\n",
    "mtp.columns=['UniProtKB', 'refseq_nt_id']\n",
    "mtp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read DisGeNet/mapa_geneid_4_uniprot_crossref.tsv for mapping\n",
    "uid_dbsnp = pd.read_csv(\"AB-15_mt-protein-UniProt_GeneName.tsv\", sep='\\t', low_memory=False, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(190964, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O60506</td>\n",
       "      <td>SYNCRIP</td>\n",
       "      <td>rs1583337818</td>\n",
       "      <td>HGVS=NC_000006.12:g.85645779_85645781del,NC_00...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O60506</td>\n",
       "      <td>SYNCRIP</td>\n",
       "      <td>rs1583337813</td>\n",
       "      <td>HGVS=NC_000006.12:g.85645775A&gt;C,NC_000006.11:g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O60506</td>\n",
       "      <td>SYNCRIP</td>\n",
       "      <td>rs1583337761</td>\n",
       "      <td>HGVS=NC_000006.12:g.85645715_85645716insGG,NC_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O60506</td>\n",
       "      <td>SYNCRIP</td>\n",
       "      <td>rs1583337683</td>\n",
       "      <td>HGVS=NC_000006.12:g.85645589A&gt;T,NC_000006.11:g...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O60506</td>\n",
       "      <td>SYNCRIP</td>\n",
       "      <td>rs1583337673</td>\n",
       "      <td>HGVS=NC_000006.12:g.85645586A&gt;C,NC_000006.11:g...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        0        1             2  \\\n",
       "0  O60506  SYNCRIP  rs1583337818   \n",
       "1  O60506  SYNCRIP  rs1583337813   \n",
       "2  O60506  SYNCRIP  rs1583337761   \n",
       "3  O60506  SYNCRIP  rs1583337683   \n",
       "4  O60506  SYNCRIP  rs1583337673   \n",
       "\n",
       "                                                   3  \n",
       "0  HGVS=NC_000006.12:g.85645779_85645781del,NC_00...  \n",
       "1  HGVS=NC_000006.12:g.85645775A>C,NC_000006.11:g...  \n",
       "2  HGVS=NC_000006.12:g.85645715_85645716insGG,NC_...  \n",
       "3  HGVS=NC_000006.12:g.85645589A>T,NC_000006.11:g...  \n",
       "4  HGVS=NC_000006.12:g.85645586A>C,NC_000006.11:g...  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(uid_dbsnp.shape, '\\n', uid_dbsnp.head())\n",
    "uid_dbsnp.columns = ['UniProtKB', 'gene_name','snpId','refseq_nt_id']\n",
    "uid_dbsnp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# find geneid for mt_proteins, rename column name of mt_proteins for merge\n",
    "mtp.columns=['UniProtKB']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge mt_proteins with DisGeNet\n",
    "mt_DGN_ug = pd.merge(mtp, DGN_ug, on='UniProtKB')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15, 2)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>GENEID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O60506</td>\n",
       "      <td>10492</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P04040</td>\n",
       "      <td>847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P04075</td>\n",
       "      <td>226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P09651</td>\n",
       "      <td>3178</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB  GENEID\n",
       "0    O60506   10492\n",
       "1    O75323    2631\n",
       "2    P04040     847\n",
       "3    P04075     226\n",
       "4    P09651    3178"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(mt_DGN_ug.shape)\n",
    "mt_DGN_ug.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to find variant for GeneID in DisGeNet, read variant-gene file\n",
    "DGN_vg = pd.read_csv(\"../DisGeNet/variant_to_gene_mappings.tsv\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(361967, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>snpId</th>\n",
       "      <th>geneId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>sourceId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs7980288</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs3832852</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs669</td>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       snpId  geneId geneSymbol sourceId\n",
       "0  rs7980288       2        A2M      VEP\n",
       "1  rs7980288       2        A2M    DBSNP\n",
       "2  rs3832852       2        A2M    DBSNP\n",
       "3      rs669       2        A2M      VEP\n",
       "4      rs669       2        A2M    DBSNP"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(DGN_vg.shape)\n",
    "DGN_vg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename column name mt_DGN_ug \n",
    "mt_DGN_ug.columns = ['UniProtKB', 'geneId']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge DisGeNet - geneId - variant\n",
    "mt_DGN_uvg = pd.merge(mt_DGN_ug, DGN_vg, on='geneId')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(550, 5) \n",
      " UniProtKB      13\n",
      "geneId         13\n",
      "snpId         280\n",
      "geneSymbol     13\n",
      "sourceId        2\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>geneId</th>\n",
       "      <th>snpId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>sourceId</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs15892</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4535700</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>DBSNP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs35844144</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4535700</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB  geneId       snpId geneSymbol sourceId\n",
       "0    O75323    2631      rs4689   NIPSNAP2      VEP\n",
       "1    O75323    2631     rs15892   NIPSNAP2      VEP\n",
       "2    O75323    2631   rs4535700   NIPSNAP2    DBSNP\n",
       "3    O75323    2631  rs35844144   NIPSNAP2      VEP\n",
       "4    O75323    2631   rs4535700   NIPSNAP2      VEP"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(mt_DGN_uvg.shape, '\\n', mt_DGN_uvg.nunique())\n",
    "mt_DGN_uvg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save file for dbSNP to RefSeq nucleotide mapping\n",
    "mt_DGN_uvg[['snpId']].to_csv('disgenet_snpId550.txt', sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read mapping file uniprot_refseq_dbsnp.txt\n",
    "# to map with rsid and uniprot\n",
    "uid_snp_nt = pd.read_csv(\"uniprot_refseq_dbsnp.txt\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(788, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "      <th>snpId</th>\n",
       "      <th>refseq_nt_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O75323</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O75323</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O75323</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O75323</td>\n",
       "      <td>NM_001483.2</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NM_001483.2:c.*251G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O75323</td>\n",
       "      <td>NM_001483.2</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NM_001483.2:c.*251G&gt;C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB    refseq_nt_id   snpId         refseq_nt_variant\n",
       "0    O75323  NM_001202469.1  rs4689  NM_001202469.1:c.*251G>A\n",
       "1    O75323  NM_001202469.1  rs4689  NM_001202469.1:c.*251G>C\n",
       "2    O75323  NM_001202469.1  rs4689  NM_001202469.1:c.*251G>T\n",
       "3    O75323     NM_001483.2  rs4689     NM_001483.2:c.*251G>A\n",
       "4    O75323     NM_001483.2  rs4689     NM_001483.2:c.*251G>C"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(uid_snp_nt.shape)\n",
    "uid_snp_nt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#merge DisGeNet - geneId - variant - uniprot - dbsnp -refseq_nt\n",
    "mt_DGN_uvg_nt = pd.merge(mt_DGN_uvg, uid_snp_nt, on=('snpId', 'UniProtKB'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1576, 7)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>UniProtKB</th>\n",
       "      <th>geneId</th>\n",
       "      <th>snpId</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>sourceId</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "      <th>refseq_nt_variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NM_001202469.1</td>\n",
       "      <td>NM_001202469.1:c.*251G&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NM_001483.2</td>\n",
       "      <td>NM_001483.2:c.*251G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O75323</td>\n",
       "      <td>2631</td>\n",
       "      <td>rs4689</td>\n",
       "      <td>NIPSNAP2</td>\n",
       "      <td>VEP</td>\n",
       "      <td>NM_001483.2</td>\n",
       "      <td>NM_001483.2:c.*251G&gt;C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  UniProtKB  geneId   snpId geneSymbol sourceId    refseq_nt_id  \\\n",
       "0    O75323    2631  rs4689   NIPSNAP2      VEP  NM_001202469.1   \n",
       "1    O75323    2631  rs4689   NIPSNAP2      VEP  NM_001202469.1   \n",
       "2    O75323    2631  rs4689   NIPSNAP2      VEP  NM_001202469.1   \n",
       "3    O75323    2631  rs4689   NIPSNAP2      VEP     NM_001483.2   \n",
       "4    O75323    2631  rs4689   NIPSNAP2      VEP     NM_001483.2   \n",
       "\n",
       "          refseq_nt_variant  \n",
       "0  NM_001202469.1:c.*251G>A  \n",
       "1  NM_001202469.1:c.*251G>C  \n",
       "2  NM_001202469.1:c.*251G>T  \n",
       "3     NM_001483.2:c.*251G>A  \n",
       "4     NM_001483.2:c.*251G>C  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(mt_DGN_uvg_nt.shape)\n",
    "mt_DGN_uvg_nt.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1576, 7) \n",
      " UniProtKB             12\n",
      "geneId                12\n",
      "snpId                219\n",
      "geneSymbol            12\n",
      "sourceId               2\n",
      "refseq_nt_id          40\n",
      "refseq_nt_variant    788\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print (mt_DGN_uvg_nt.shape, '\\n', mt_DGN_uvg_nt.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_DGN_uvg_nt = mt_DGN_uvg_nt.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_DGN_uvg_nt.to_csv('AB15_mt_DGN_uvg_nt.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!head  AB15_mt_DGN_uvg_nt.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read clinvar_pathogenic_2805_40uid.txt | clinvar_pathogenic_2805.txt | clinvar_pathogenic_NM_259050.tsv| variant_summary.txt\n",
    "CV_path = pd.read_csv(\"../ClinVar/variant_summary.txt\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1656032, 34)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>ReviewStatus</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215070,OMIM:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>4820844</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215070,OMIM:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>4781213</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215072,OMIM:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4827360</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215072,OMIM:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4787729</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,O...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>85342440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #AlleleID                       Type  \\\n",
       "0      15041                      Indel   \n",
       "1      15041                      Indel   \n",
       "2      15042                   Deletion   \n",
       "3      15042                   Deletion   \n",
       "4      15043  single nucleotide variant   \n",
       "\n",
       "                                                Name  GeneID GeneSymbol  \\\n",
       "0  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "1  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "2     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "3     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "4       NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n",
       "\n",
       "      HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "0  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "1  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "2  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "3  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "4  HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n",
       "\n",
       "   RS# (dbSNP)  ...                    ReviewStatus NumberSubmitters  \\\n",
       "0    397704705  ...  no assertion criteria provided                1   \n",
       "1    397704705  ...  no assertion criteria provided                1   \n",
       "2    397704709  ...  no assertion criteria provided                1   \n",
       "3    397704709  ...  no assertion criteria provided                1   \n",
       "4    150829393  ...  no assertion criteria provided                1   \n",
       "\n",
       "  Guidelines TestedInGTR                                           OtherIDs  \\\n",
       "0          -           N                  ClinGen:CA215070,OMIM:613653.0001   \n",
       "1          -           N                  ClinGen:CA215070,OMIM:613653.0001   \n",
       "2          -           N                  ClinGen:CA215072,OMIM:613653.0002   \n",
       "3          -           N                  ClinGen:CA215072,OMIM:613653.0002   \n",
       "4          -           N  ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,O...   \n",
       "\n",
       "  SubmitterCategories VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "0                   1           2     4820844               GGAT   \n",
       "1                   1           2     4781213               GGAT   \n",
       "2                   1           3     4827360    GCTGCTGGACCTGCC   \n",
       "3                   1           3     4787729    GCTGCTGGACCTGCC   \n",
       "4                   1           4    85342440                  G   \n",
       "\n",
       "       AlternateAlleleVCF  \n",
       "0  TGCTGTAAACTGTAACTGTAAA  \n",
       "1  TGCTGTAAACTGTAACTGTAAA  \n",
       "2                       G  \n",
       "3                       G  \n",
       "4                       A  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#mt_CV.nunique()\n",
    "print(CV_path.shape)\n",
    "CV_path.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "CV_path['refseq_nt_id'] = CV_path['Name'].str.split('(',expand=True)[0].str.split(':',expand=True)[0].str.rstrip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215070,OMIM:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>4820844</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>NM_014855.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215070,OMIM:613653.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>4781213</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>NM_014855.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215072,OMIM:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4827360</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_014855.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA215072,OMIM:613653.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>4787729</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "      <td>NM_014855.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,O...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>85342440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NM_014630.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #AlleleID                       Type  \\\n",
       "0      15041                      Indel   \n",
       "1      15041                      Indel   \n",
       "2      15042                   Deletion   \n",
       "3      15042                   Deletion   \n",
       "4      15043  single nucleotide variant   \n",
       "\n",
       "                                                Name  GeneID GeneSymbol  \\\n",
       "0  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "1  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "2     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "3     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "4       NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n",
       "\n",
       "      HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "0  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "1  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "2  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "3  HGNC:22197              Pathogenic              1  Jun 29, 2010   \n",
       "4  HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n",
       "\n",
       "   RS# (dbSNP)  ... NumberSubmitters Guidelines TestedInGTR  \\\n",
       "0    397704705  ...                1          -           N   \n",
       "1    397704705  ...                1          -           N   \n",
       "2    397704709  ...                1          -           N   \n",
       "3    397704709  ...                1          -           N   \n",
       "4    150829393  ...                1          -           N   \n",
       "\n",
       "                                            OtherIDs SubmitterCategories  \\\n",
       "0                  ClinGen:CA215070,OMIM:613653.0001                   1   \n",
       "1                  ClinGen:CA215070,OMIM:613653.0001                   1   \n",
       "2                  ClinGen:CA215072,OMIM:613653.0002                   1   \n",
       "3                  ClinGen:CA215072,OMIM:613653.0002                   1   \n",
       "4  ClinGen:CA210674,UniProtKB:Q92610#VAR_064583,O...                   1   \n",
       "\n",
       "  VariationID PositionVCF ReferenceAlleleVCF      AlternateAlleleVCF  \\\n",
       "0           2     4820844               GGAT  TGCTGTAAACTGTAACTGTAAA   \n",
       "1           2     4781213               GGAT  TGCTGTAAACTGTAACTGTAAA   \n",
       "2           3     4827360    GCTGCTGGACCTGCC                       G   \n",
       "3           3     4787729    GCTGCTGGACCTGCC                       G   \n",
       "4           4    85342440                  G                       A   \n",
       "\n",
       "   refseq_nt_id  \n",
       "0   NM_014855.3  \n",
       "1   NM_014855.3  \n",
       "2   NM_014855.3  \n",
       "3   NM_014855.3  \n",
       "4   NM_014630.3  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CV_path.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1656032, 35)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CV_path.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>15316</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>5'UTR-exon 1 deletion (2703 bp)</td>\n",
       "      <td>124583</td>\n",
       "      <td>CANT1</td>\n",
       "      <td>HGNC:19721</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 01, 2009</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv7487213,OMIM:613165.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>277</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>5'UTR-exon 1 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1570</th>\n",
       "      <td>15873</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>APC deletion</td>\n",
       "      <td>324</td>\n",
       "      <td>APC</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 08, 1999</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>ACMG2013,ACMG2016</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611731.0042</td>\n",
       "      <td>1</td>\n",
       "      <td>834</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>APC deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1571</th>\n",
       "      <td>15873</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>APC deletion</td>\n",
       "      <td>324</td>\n",
       "      <td>APC</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 08, 1999</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>ACMG2013,ACMG2016</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611731.0042</td>\n",
       "      <td>1</td>\n",
       "      <td>834</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>APC deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5103</th>\n",
       "      <td>17795</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>LEMD3, 1941+5delG</td>\n",
       "      <td>23592</td>\n",
       "      <td>LEMD3</td>\n",
       "      <td>HGNC:28887</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 01, 2004</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:607844.0005</td>\n",
       "      <td>1</td>\n",
       "      <td>2756</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>LEMD3, 1941+5delG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9278</th>\n",
       "      <td>20090</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>ARFGEF2, PRO81GLN, VAL83LEU, 249delA</td>\n",
       "      <td>10564</td>\n",
       "      <td>ARFGEF2</td>\n",
       "      <td>HGNC:15853</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jan 01, 2004</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:605371.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>5051</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ARFGEF2, PRO81GLN, VAL83LEU, 249delA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863679</th>\n",
       "      <td>553338</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>7q31.32deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562164</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>7q31.32deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863684</th>\n",
       "      <td>553343</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>9p24.1deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562169</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>9p24.1deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863691</th>\n",
       "      <td>553350</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>22q11.21deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562176</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>22q11.21deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320787</th>\n",
       "      <td>800670</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>15q22.2deletion</td>\n",
       "      <td>6095</td>\n",
       "      <td>RORA</td>\n",
       "      <td>HGNC:10258</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 11, 2019</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>812183</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>15q22.2deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1323283</th>\n",
       "      <td>802083</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>15q15.3 deletion</td>\n",
       "      <td>161497</td>\n",
       "      <td>STRC</td>\n",
       "      <td>HGNC:16035</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 24, 2019</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>813830</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>15q15.3 deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID      Type                                  Name  GeneID  \\\n",
       "534          15316  Deletion       5'UTR-exon 1 deletion (2703 bp)  124583   \n",
       "1570         15873  Deletion                          APC deletion     324   \n",
       "1571         15873  Deletion                          APC deletion     324   \n",
       "5103         17795  Deletion                     LEMD3, 1941+5delG   23592   \n",
       "9278         20090  Deletion  ARFGEF2, PRO81GLN, VAL83LEU, 249delA   10564   \n",
       "...            ...       ...                                   ...     ...   \n",
       "863679      553338  Deletion                       7q31.32deletion      -1   \n",
       "863684      553343  Deletion                        9p24.1deletion      -1   \n",
       "863691      553350  Deletion                      22q11.21deletion      -1   \n",
       "1320787     800670  Deletion                       15q22.2deletion    6095   \n",
       "1323283     802083  Deletion                      15q15.3 deletion  161497   \n",
       "\n",
       "        GeneSymbol     HGNC_ID ClinicalSignificance  ClinSigSimple  \\\n",
       "534          CANT1  HGNC:19721           Pathogenic              1   \n",
       "1570           APC    HGNC:583           Pathogenic              1   \n",
       "1571           APC    HGNC:583           Pathogenic              1   \n",
       "5103         LEMD3  HGNC:28887           Pathogenic              1   \n",
       "9278       ARFGEF2  HGNC:15853           Pathogenic              1   \n",
       "...            ...         ...                  ...            ...   \n",
       "863679           -           -           Pathogenic              1   \n",
       "863684           -           -           Pathogenic              1   \n",
       "863691           -           -           Pathogenic              1   \n",
       "1320787       RORA  HGNC:10258    Likely pathogenic              1   \n",
       "1323283       STRC  HGNC:16035           Pathogenic              1   \n",
       "\n",
       "        LastEvaluated  RS# (dbSNP)  ... NumberSubmitters         Guidelines  \\\n",
       "534      Nov 01, 2009           -1  ...                1                  -   \n",
       "1570     Oct 08, 1999           -1  ...                1  ACMG2013,ACMG2016   \n",
       "1571     Oct 08, 1999           -1  ...                1  ACMG2013,ACMG2016   \n",
       "5103     Nov 01, 2004           -1  ...                1                  -   \n",
       "9278     Jan 01, 2004           -1  ...                1                  -   \n",
       "...               ...          ...  ...              ...                ...   \n",
       "863679   Aug 01, 2018           -1  ...                1                  -   \n",
       "863684   Aug 01, 2018           -1  ...                1                  -   \n",
       "863691   Aug 01, 2018           -1  ...                1                  -   \n",
       "1320787  Dec 11, 2019           -1  ...                1                  -   \n",
       "1323283  Nov 24, 2019           -1  ...                1                  -   \n",
       "\n",
       "        TestedInGTR                            OtherIDs SubmitterCategories  \\\n",
       "534               N  dbVar:nssv7487213,OMIM:613165.0001                   1   \n",
       "1570              N                    OMIM:611731.0042                   1   \n",
       "1571              N                    OMIM:611731.0042                   1   \n",
       "5103              N                    OMIM:607844.0005                   1   \n",
       "9278              N                    OMIM:605371.0002                   1   \n",
       "...             ...                                 ...                 ...   \n",
       "863679            N                                   -                   2   \n",
       "863684            N                                   -                   2   \n",
       "863691            N                                   -                   2   \n",
       "1320787           N                                   -                   2   \n",
       "1323283           N                                   -                   2   \n",
       "\n",
       "        VariationID PositionVCF ReferenceAlleleVCF AlternateAlleleVCF  \\\n",
       "534             277          -1                 na                 na   \n",
       "1570            834          -1                 na                 na   \n",
       "1571            834          -1                 na                 na   \n",
       "5103           2756          -1                 na                 na   \n",
       "9278           5051          -1                 na                 na   \n",
       "...             ...         ...                ...                ...   \n",
       "863679       562164          -1                 na                 na   \n",
       "863684       562169          -1                 na                 na   \n",
       "863691       562176          -1                 na                 na   \n",
       "1320787      812183          -1                 na                 na   \n",
       "1323283      813830          -1                 na                 na   \n",
       "\n",
       "                                 refseq_nt_id  \n",
       "534                     5'UTR-exon 1 deletion  \n",
       "1570                             APC deletion  \n",
       "1571                             APC deletion  \n",
       "5103                        LEMD3, 1941+5delG  \n",
       "9278     ARFGEF2, PRO81GLN, VAL83LEU, 249delA  \n",
       "...                                       ...  \n",
       "863679                        7q31.32deletion  \n",
       "863684                         9p24.1deletion  \n",
       "863691                       22q11.21deletion  \n",
       "1320787                       15q22.2deletion  \n",
       "1323283                      15q15.3 deletion  \n",
       "\n",
       "[95 rows x 35 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CV_path[CV_path['refseq_nt_id'].str.contains(\"del\", na=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "CV_path['refseq_nt_id']=CV_path['refseq_nt_id'].replace(to_replace =\"NM_005881.3c.222delG\", value =\"NM_005881.3\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>NumberSubmitters</th>\n",
       "      <th>Guidelines</th>\n",
       "      <th>TestedInGTR</th>\n",
       "      <th>OtherIDs</th>\n",
       "      <th>SubmitterCategories</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>refseq_nt_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>15316</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>5'UTR-exon 1 deletion (2703 bp)</td>\n",
       "      <td>124583</td>\n",
       "      <td>CANT1</td>\n",
       "      <td>HGNC:19721</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 01, 2009</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>dbVar:nssv7487213,OMIM:613165.0001</td>\n",
       "      <td>1</td>\n",
       "      <td>277</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>5'UTR-exon 1 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1570</th>\n",
       "      <td>15873</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>APC deletion</td>\n",
       "      <td>324</td>\n",
       "      <td>APC</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 08, 1999</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>ACMG2013,ACMG2016</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611731.0042</td>\n",
       "      <td>1</td>\n",
       "      <td>834</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>APC deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1571</th>\n",
       "      <td>15873</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>APC deletion</td>\n",
       "      <td>324</td>\n",
       "      <td>APC</td>\n",
       "      <td>HGNC:583</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Oct 08, 1999</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>ACMG2013,ACMG2016</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:611731.0042</td>\n",
       "      <td>1</td>\n",
       "      <td>834</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>APC deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5103</th>\n",
       "      <td>17795</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>LEMD3, 1941+5delG</td>\n",
       "      <td>23592</td>\n",
       "      <td>LEMD3</td>\n",
       "      <td>HGNC:28887</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 01, 2004</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:607844.0005</td>\n",
       "      <td>1</td>\n",
       "      <td>2756</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>LEMD3, 1941+5delG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9278</th>\n",
       "      <td>20090</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>ARFGEF2, PRO81GLN, VAL83LEU, 249delA</td>\n",
       "      <td>10564</td>\n",
       "      <td>ARFGEF2</td>\n",
       "      <td>HGNC:15853</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Jan 01, 2004</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>OMIM:605371.0002</td>\n",
       "      <td>1</td>\n",
       "      <td>5051</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>ARFGEF2, PRO81GLN, VAL83LEU, 249delA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863679</th>\n",
       "      <td>553338</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>7q31.32deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562164</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>7q31.32deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863684</th>\n",
       "      <td>553343</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>9p24.1deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562169</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>9p24.1deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>863691</th>\n",
       "      <td>553350</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>22q11.21deletion</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Aug 01, 2018</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>562176</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>22q11.21deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320787</th>\n",
       "      <td>800670</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>15q22.2deletion</td>\n",
       "      <td>6095</td>\n",
       "      <td>RORA</td>\n",
       "      <td>HGNC:10258</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Dec 11, 2019</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>812183</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>15q22.2deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1323283</th>\n",
       "      <td>802083</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>15q15.3 deletion</td>\n",
       "      <td>161497</td>\n",
       "      <td>STRC</td>\n",
       "      <td>HGNC:16035</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1</td>\n",
       "      <td>Nov 24, 2019</td>\n",
       "      <td>-1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>N</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>813830</td>\n",
       "      <td>-1</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>15q15.3 deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>94 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         #AlleleID      Type                                  Name  GeneID  \\\n",
       "534          15316  Deletion       5'UTR-exon 1 deletion (2703 bp)  124583   \n",
       "1570         15873  Deletion                          APC deletion     324   \n",
       "1571         15873  Deletion                          APC deletion     324   \n",
       "5103         17795  Deletion                     LEMD3, 1941+5delG   23592   \n",
       "9278         20090  Deletion  ARFGEF2, PRO81GLN, VAL83LEU, 249delA   10564   \n",
       "...            ...       ...                                   ...     ...   \n",
       "863679      553338  Deletion                       7q31.32deletion      -1   \n",
       "863684      553343  Deletion                        9p24.1deletion      -1   \n",
       "863691      553350  Deletion                      22q11.21deletion      -1   \n",
       "1320787     800670  Deletion                       15q22.2deletion    6095   \n",
       "1323283     802083  Deletion                      15q15.3 deletion  161497   \n",
       "\n",
       "        GeneSymbol     HGNC_ID ClinicalSignificance  ClinSigSimple  \\\n",
       "534          CANT1  HGNC:19721           Pathogenic              1   \n",
       "1570           APC    HGNC:583           Pathogenic              1   \n",
       "1571           APC    HGNC:583           Pathogenic              1   \n",
       "5103         LEMD3  HGNC:28887           Pathogenic              1   \n",
       "9278       ARFGEF2  HGNC:15853           Pathogenic              1   \n",
       "...            ...         ...                  ...            ...   \n",
       "863679           -           -           Pathogenic              1   \n",
       "863684           -           -           Pathogenic              1   \n",
       "863691           -           -           Pathogenic              1   \n",
       "1320787       RORA  HGNC:10258    Likely pathogenic              1   \n",
       "1323283       STRC  HGNC:16035           Pathogenic              1   \n",
       "\n",
       "        LastEvaluated  RS# (dbSNP)  ... NumberSubmitters         Guidelines  \\\n",
       "534      Nov 01, 2009           -1  ...                1                  -   \n",
       "1570     Oct 08, 1999           -1  ...                1  ACMG2013,ACMG2016   \n",
       "1571     Oct 08, 1999           -1  ...                1  ACMG2013,ACMG2016   \n",
       "5103     Nov 01, 2004           -1  ...                1                  -   \n",
       "9278     Jan 01, 2004           -1  ...                1                  -   \n",
       "...               ...          ...  ...              ...                ...   \n",
       "863679   Aug 01, 2018           -1  ...                1                  -   \n",
       "863684   Aug 01, 2018           -1  ...                1                  -   \n",
       "863691   Aug 01, 2018           -1  ...                1                  -   \n",
       "1320787  Dec 11, 2019           -1  ...                1                  -   \n",
       "1323283  Nov 24, 2019           -1  ...                1                  -   \n",
       "\n",
       "        TestedInGTR                            OtherIDs SubmitterCategories  \\\n",
       "534               N  dbVar:nssv7487213,OMIM:613165.0001                   1   \n",
       "1570              N                    OMIM:611731.0042                   1   \n",
       "1571              N                    OMIM:611731.0042                   1   \n",
       "5103              N                    OMIM:607844.0005                   1   \n",
       "9278              N                    OMIM:605371.0002                   1   \n",
       "...             ...                                 ...                 ...   \n",
       "863679            N                                   -                   2   \n",
       "863684            N                                   -                   2   \n",
       "863691            N                                   -                   2   \n",
       "1320787           N                                   -                   2   \n",
       "1323283           N                                   -                   2   \n",
       "\n",
       "        VariationID PositionVCF ReferenceAlleleVCF AlternateAlleleVCF  \\\n",
       "534             277          -1                 na                 na   \n",
       "1570            834          -1                 na                 na   \n",
       "1571            834          -1                 na                 na   \n",
       "5103           2756          -1                 na                 na   \n",
       "9278           5051          -1                 na                 na   \n",
       "...             ...         ...                ...                ...   \n",
       "863679       562164          -1                 na                 na   \n",
       "863684       562169          -1                 na                 na   \n",
       "863691       562176          -1                 na                 na   \n",
       "1320787      812183          -1                 na                 na   \n",
       "1323283      813830          -1                 na                 na   \n",
       "\n",
       "                                 refseq_nt_id  \n",
       "534                     5'UTR-exon 1 deletion  \n",
       "1570                             APC deletion  \n",
       "1571                             APC deletion  \n",
       "5103                        LEMD3, 1941+5delG  \n",
       "9278     ARFGEF2, PRO81GLN, VAL83LEU, 249delA  \n",
       "...                                       ...  \n",
       "863679                        7q31.32deletion  \n",
       "863684                         9p24.1deletion  \n",
       "863691                       22q11.21deletion  \n",
       "1320787                       15q22.2deletion  \n",
       "1323283                      15q15.3 deletion  \n",
       "\n",
       "[94 rows x 35 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CV_path[CV_path['refseq_nt_id'].str.contains(\"del\", na=False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read UniProt - RefSeq_nt file AB-15_mt-protein-UniProt_RefSeq-nt.txt\n",
    "uid_nt= pd.read_csv(\"AB-15_mt-protein-UniProt_RefSeq-nt.txt\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "uid_nt.columns=['UniProtKB', 'refseq_nt_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    " mt_uid_nt_CV = pd.merge(uid_nt, CV_path, on='refseq_nt_id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(244, 36) \n",
      " UniProtKB                 5\n",
      "refseq_nt_id              7\n",
      "#AlleleID               122\n",
      "Type                      6\n",
      "Name                    122\n",
      "GeneID                    5\n",
      "GeneSymbol                5\n",
      "HGNC_ID                   5\n",
      "ClinicalSignificance      9\n",
      "ClinSigSimple             2\n",
      "LastEvaluated            60\n",
      "RS# (dbSNP)             117\n",
      "nsv/esv (dbVar)           1\n",
      "RCVaccession            122\n",
      "PhenotypeIDS             23\n",
      "PhenotypeList            23\n",
      "Origin                    5\n",
      "OriginSimple              2\n",
      "Assembly                  2\n",
      "ChromosomeAccession       8\n",
      "Chromosome                4\n",
      "Start                   236\n",
      "Stop                    238\n",
      "ReferenceAllele           1\n",
      "AlternateAllele           1\n",
      "Cytogenetic               5\n",
      "ReviewStatus              5\n",
      "NumberSubmitters          5\n",
      "Guidelines                2\n",
      "TestedInGTR               1\n",
      "OtherIDs                105\n",
      "SubmitterCategories       3\n",
      "VariationID             122\n",
      "PositionVCF             231\n",
      "ReferenceAlleleVCF       13\n",
      "AlternateAlleleVCF       12\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(mt_uid_nt_CV.shape, '\\n', mt_uid_nt_CV.nunique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/rakesh/Host-Pathogen_Interactions_in_Mitochondrial_and_Infectious_Diseases/AB15_dataset/UniProt_dbSNP\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
