{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.1.3'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# mapping MSeqDR trasncript to mutation and diseases\n",
    "import pandas as pd\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1780, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI Protein ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_000007.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_000010.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_000017.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000024.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_000039.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  NCBI Protein ID\n",
       "0     NP_000007.1\n",
       "1     NP_000010.1\n",
       "2     NP_000017.1\n",
       "3     NP_000024.2\n",
       "4     NP_000039.2"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read transcript\n",
    "ncbi_id= pd.read_csv(\"NCBI_Protein_ID_uniprot_ids_5778_tr_ids1788_c1_1780.txt\", sep='\\t', names=[\"NCBI Protein ID\"])\n",
    "print(ncbi_id.shape)\n",
    "ncbi_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37168, 32)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Name</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>NCBI Protein ID</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Type</th>\n",
       "      <th>DNA change (genomic) (hg19)</th>\n",
       "      <th>Published as</th>\n",
       "      <th>GERP</th>\n",
       "      <th>...</th>\n",
       "      <th>DNA change (cDNA)</th>\n",
       "      <th>Protein</th>\n",
       "      <th>GVS function</th>\n",
       "      <th>Splice distance</th>\n",
       "      <th>Position</th>\n",
       "      <th>Location</th>\n",
       "      <th>Exon</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>RNA change</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>NM_020745.3</td>\n",
       "      <td>NP_065796.1</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.44266473G&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*1811C&gt;T</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>NM_020745.3</td>\n",
       "      <td>NP_065796.1</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.44266564G&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*1720C&gt;T</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>NM_020745.3</td>\n",
       "      <td>NP_065796.1</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.44266565A&gt;G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*1719T&gt;C</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>NM_020745.3</td>\n",
       "      <td>NP_065796.1</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.44266574C&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*1710G&gt;T</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>NM_020745.3</td>\n",
       "      <td>NP_065796.1</td>\n",
       "      <td>6</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.44266578G&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.*1706C&gt;T</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>r.(=)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene                                     Name   Transcript  \\\n",
       "0  AARS2  alanyl-tRNA synthetase 2, mitochondrial  NM_020745.3   \n",
       "1  AARS2  alanyl-tRNA synthetase 2, mitochondrial  NM_020745.3   \n",
       "2  AARS2  alanyl-tRNA synthetase 2, mitochondrial  NM_020745.3   \n",
       "3  AARS2  alanyl-tRNA synthetase 2, mitochondrial  NM_020745.3   \n",
       "4  AARS2  alanyl-tRNA synthetase 2, mitochondrial  NM_020745.3   \n",
       "\n",
       "  NCBI Protein ID Chr   Allele Type DNA change (genomic) (hg19) Published as  \\\n",
       "0     NP_065796.1   6  Unknown    -               g.44266473G>A            -   \n",
       "1     NP_065796.1   6  Unknown    -               g.44266564G>A            -   \n",
       "2     NP_065796.1   6  Unknown    -               g.44266565A>G            -   \n",
       "3     NP_065796.1   6  Unknown    -               g.44266574C>A            -   \n",
       "4     NP_065796.1   6  Unknown    -               g.44266578G>A            -   \n",
       "\n",
       "  GERP  ... DNA change (cDNA) Protein GVS function Splice distance Position  \\\n",
       "0    -  ...        c.*1811C>T   p.(=)            -               -        -   \n",
       "1    -  ...        c.*1720C>T   p.(=)            -               -        -   \n",
       "2    -  ...        c.*1719T>C   p.(=)            -               -        -   \n",
       "3    -  ...        c.*1710G>T   p.(=)            -               -        -   \n",
       "4    -  ...        c.*1706C>T   p.(=)            -               -        -   \n",
       "\n",
       "  Location Exon PolyPhen RNA change SIFT  \n",
       "0        -    -        -      r.(=)    -  \n",
       "1        -    -        -      r.(=)    -  \n",
       "2        -    -        -      r.(=)    -  \n",
       "3        -    -        -      r.(=)    -  \n",
       "4        -    -        -      r.(=)    -  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read all variants\n",
    "variants= pd.read_csv(\"MSeqDR_all_variants_transcripts_data.txt\", sep='\\t', low_memory=False)\n",
    "print(variants.shape)\n",
    "variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(18976, 32)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NCBI Protein ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Name</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Type</th>\n",
       "      <th>DNA change (genomic) (hg19)</th>\n",
       "      <th>Published as</th>\n",
       "      <th>GERP</th>\n",
       "      <th>...</th>\n",
       "      <th>DNA change (cDNA)</th>\n",
       "      <th>Protein</th>\n",
       "      <th>GVS function</th>\n",
       "      <th>Splice distance</th>\n",
       "      <th>Position</th>\n",
       "      <th>Location</th>\n",
       "      <th>Exon</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>RNA change</th>\n",
       "      <th>SIFT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NP_000007.1</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>NM_000016.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.76194174G&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.118+1G&gt;T</td>\n",
       "      <td>p.?</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NP_000007.1</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>NM_000016.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.76194191T&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.118+18T&gt;A</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NP_000007.1</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>NM_000016.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.76194230del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.118+57del</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NP_000007.1</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>NM_000016.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.76194235del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.118+62del</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NP_000007.1</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>transcript variant 1</td>\n",
       "      <td>NM_000016.4</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>-</td>\n",
       "      <td>g.76194287A&gt;G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>c.118+114A&gt;G</td>\n",
       "      <td>p.(=)</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  NCBI Protein ID   Gene                  Name   Transcript Chr   Allele Type  \\\n",
       "0     NP_000007.1  ACADM  transcript variant 1  NM_000016.4   1  Unknown    -   \n",
       "1     NP_000007.1  ACADM  transcript variant 1  NM_000016.4   1  Unknown    -   \n",
       "2     NP_000007.1  ACADM  transcript variant 1  NM_000016.4   1  Unknown    -   \n",
       "3     NP_000007.1  ACADM  transcript variant 1  NM_000016.4   1  Unknown    -   \n",
       "4     NP_000007.1  ACADM  transcript variant 1  NM_000016.4   1  Unknown    -   \n",
       "\n",
       "  DNA change (genomic) (hg19) Published as GERP  ... DNA change (cDNA)  \\\n",
       "0               g.76194174G>T            -    -  ...        c.118+1G>T   \n",
       "1               g.76194191T>A            -    -  ...       c.118+18T>A   \n",
       "2               g.76194230del            -    -  ...       c.118+57del   \n",
       "3               g.76194235del            -    -  ...       c.118+62del   \n",
       "4               g.76194287A>G            -    -  ...      c.118+114A>G   \n",
       "\n",
       "  Protein GVS function Splice distance Position Location Exon PolyPhen  \\\n",
       "0     p.?            -               -        -        -    -        -   \n",
       "1   p.(=)            -               -        -        -    -        -   \n",
       "2   p.(=)            -               -        -        -    -        -   \n",
       "3   p.(=)            -               -        -        -    -        -   \n",
       "4   p.(=)            -               -        -        -    -        -   \n",
       "\n",
       "  RNA change SIFT  \n",
       "0          -    -  \n",
       "1          -    -  \n",
       "2          -    -  \n",
       "3          -    -  \n",
       "4          -    -  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# inner join on 'NCBI Protein ID'\n",
    "variants_ncbi_id= pd.merge(ncbi_id, variants, on='NCBI Protein ID')\n",
    "variants_ncbi_id.to_csv(\"variants18976_ncbi_id1780.tsv\", sep=\"\\t\", index=False)\n",
    "print(variants_ncbi_id.shape)\n",
    "variants_ncbi_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(284, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACADM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPT2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Symbol\n",
       "0  ACADM\n",
       "1  ACAT1\n",
       "2   ASS1\n",
       "3  BRCA2\n",
       "4   CPT2"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#find unique genes\n",
    "variants_ncbi_id.Gene.unique()\n",
    "variants_ncbi_id.Gene.nunique()\n",
    "variants_genes = pd.DataFrame(variants_ncbi_id.Gene.unique())\n",
    "variants_genes.columns=['Symbol']\n",
    "print(variants_genes.shape)\n",
    "variants_genes.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes= pd.read_csv(\"MSeqDR_all_genes_to_disease.txt\", sep='\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1627, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Band</th>\n",
       "      <th>Transcripts</th>\n",
       "      <th>Variants</th>\n",
       "      <th>Unique variants</th>\n",
       "      <th>Last updated</th>\n",
       "      <th>Associated with diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>aminoadipate aminotransferase</td>\n",
       "      <td>4</td>\n",
       "      <td>q33</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>alanyl-tRNA synthetase 2, mitochondrial</td>\n",
       "      <td>6</td>\n",
       "      <td>p21.1</td>\n",
       "      <td>3</td>\n",
       "      <td>93</td>\n",
       "      <td>90</td>\n",
       "      <td>13-11-2013</td>\n",
       "      <td>COXPD8, LKENP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AASS</td>\n",
       "      <td>aminoadipate-semialdehyde synthase</td>\n",
       "      <td>7</td>\n",
       "      <td>q31.3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABAT</td>\n",
       "      <td>4-aminobutyrate aminotransferase</td>\n",
       "      <td>16</td>\n",
       "      <td>p13.2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABCA13</td>\n",
       "      <td>ATP binding cassette subfamily A member 13</td>\n",
       "      <td>7</td>\n",
       "      <td>p12.3</td>\n",
       "      <td>29</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Symbol                                        Gene Chr   Band  Transcripts  \\\n",
       "0   AADAT               aminoadipate aminotransferase   4    q33            4   \n",
       "1   AARS2     alanyl-tRNA synthetase 2, mitochondrial   6  p21.1            3   \n",
       "2    AASS          aminoadipate-semialdehyde synthase   7  q31.3            1   \n",
       "3    ABAT            4-aminobutyrate aminotransferase  16  p13.2            3   \n",
       "4  ABCA13  ATP binding cassette subfamily A member 13   7  p12.3           29   \n",
       "\n",
       "   Variants  Unique variants Last updated Associated with diseases  \n",
       "0         0                0          NaN                        -  \n",
       "1        93               90   13-11-2013            COXPD8, LKENP  \n",
       "2         0                0          NaN                        -  \n",
       "3         0                0          NaN                        -  \n",
       "4         0                0          NaN                        -  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(genes.shape)\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(284, 9)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Band</th>\n",
       "      <th>Transcripts</th>\n",
       "      <th>Variants</th>\n",
       "      <th>Unique variants</th>\n",
       "      <th>Last updated</th>\n",
       "      <th>Associated with diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ACADM</td>\n",
       "      <td>acyl-CoA dehydrogenase, C-4 to C-12 straight c...</td>\n",
       "      <td>1</td>\n",
       "      <td>p31</td>\n",
       "      <td>2</td>\n",
       "      <td>217</td>\n",
       "      <td>205</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACADMD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ACAT1</td>\n",
       "      <td>acetyl-CoA acetyltransferase 1</td>\n",
       "      <td>11</td>\n",
       "      <td>q22.3</td>\n",
       "      <td>1</td>\n",
       "      <td>95</td>\n",
       "      <td>85</td>\n",
       "      <td>NaN</td>\n",
       "      <td>203750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASS1</td>\n",
       "      <td>argininosuccinate synthase 1</td>\n",
       "      <td>9</td>\n",
       "      <td>q34.1</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>breast cancer 2, early onset</td>\n",
       "      <td>13</td>\n",
       "      <td>q12-q13</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>15-09-2014</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CPT2</td>\n",
       "      <td>carnitine palmitoyltransferase 2</td>\n",
       "      <td>1</td>\n",
       "      <td>p32</td>\n",
       "      <td>1</td>\n",
       "      <td>100</td>\n",
       "      <td>87</td>\n",
       "      <td>NaN</td>\n",
       "      <td>255110, 608836, CPT II</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Symbol                                               Gene Chr     Band  \\\n",
       "0  ACADM  acyl-CoA dehydrogenase, C-4 to C-12 straight c...   1      p31   \n",
       "1  ACAT1                     acetyl-CoA acetyltransferase 1  11    q22.3   \n",
       "2   ASS1                       argininosuccinate synthase 1   9    q34.1   \n",
       "3  BRCA2                       breast cancer 2, early onset  13  q12-q13   \n",
       "4   CPT2                   carnitine palmitoyltransferase 2   1      p32   \n",
       "\n",
       "   Transcripts  Variants  Unique variants Last updated  \\\n",
       "0            2       217              205          NaN   \n",
       "1            1        95               85          NaN   \n",
       "2            2        15               15          NaN   \n",
       "3            1         1                1   15-09-2014   \n",
       "4            1       100               87          NaN   \n",
       "\n",
       "  Associated with diseases  \n",
       "0                   ACADMD  \n",
       "1                   203750  \n",
       "2                        -  \n",
       "3                        -  \n",
       "4   255110, 608836, CPT II  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get mapped genes to get list of diseases\n",
    "# inner join on 'Symbol'\n",
    "genes_variants_ncbi_id= pd.merge(variants_genes, genes, on='Symbol')\n",
    "genes_variants_ncbi_id.to_csv(\"gene284_variants18976_ncbi_id1780.tsv\", sep=\"\\t\", index=False)\n",
    "print(genes_variants_ncbi_id.shape)\n",
    "genes_variants_ncbi_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#find unique Associated with diseases\n",
    "\"\"\"variants_ncbi_id.Associated with diseases.unique()\n",
    "variants_ncbi_id.Gene.nunique()\n",
    "variants_genes = pd.DataFrame(variants_ncbi_id.Gene.unique())\n",
    "variants_genes.columns=['Symbol']\n",
    "print(variants_genes.shape)\n",
    "variants_genes.head()\"\"\"\n",
    "\n",
    "genes_diseaseAbbr = pd.DataFrame(genes_variants_ncbi_id.iloc[:, 8].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(149, 1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Abbreviation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[ACADMD]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[203750]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[-]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[255110,  608836,  CPT II]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[233690]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>[SNDI]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>[252010,  MC1DN10 618233]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>[616370]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>[252010,  618240]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>[615578]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>149 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Abbreviation\n",
       "0                      [ACADMD]\n",
       "1                      [203750]\n",
       "2                           [-]\n",
       "3    [255110,  608836,  CPT II]\n",
       "4                      [233690]\n",
       "..                          ...\n",
       "144                      [SNDI]\n",
       "145   [252010,  MC1DN10 618233]\n",
       "146                    [616370]\n",
       "147           [252010,  618240]\n",
       "148                    [615578]\n",
       "\n",
       "[149 rows x 1 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_diseaseAbbr.columns = ['Abbreviation']\n",
    "print(genes_diseaseAbbr.shape)\n",
    "genes_diseaseAbbr.head()\n",
    "genes_diseaseAbbr.assign(Abbreviation=genes_diseaseAbbr.Abbreviation.str.split(','))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "pandas.core.frame.DataFrame"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(genes_diseaseAbbr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
